### DISCUSSION TITLE
Some problem about the focus in #2724.

### DISCUSSION THREAD

**Discussion by BernieHuang2008** [searxng/searxng#2736] [General]
https://github.com/searxng/searxng/assets/88757735/efebe378-fde2-4547-8d43-e3fbc48a05ec

this is a problem of focus.

**Comment by BernieHuang2008** [comment]
pseudo code for user's operation to explain this:
>
>1. `search & choose a image result.`
-> that maens the focus point is on `.image-detail-open > .result`
=> we can detect it when binding on `.image-detail-open`.
>
>2. `click the blank of preview window`
-> this will change the focus point to prewiew window (`.detail`)
=> this time, when i press the arrow button now, the handler binding on `.image-detail-open` won't be actived because it doesn't have the focus.
>
>So that's the problem appeared.

**Comment by bonswouar** [comment]
Ok I get it now! Sorry I missed the step of clicking on the "blank part".

I'm not a js expert, any advice would be welcome!
I've tried to set the selector to `'.result-images .detail, .image-detail-open'` but for some reason the first part doesn't work (although I'd think the focus would be on `.detail` when clicking on a blank part of this div)

**Reply by BernieHuang2008** [reply]
i got the same result with you: think it would be on `.detail` but actually not.

**Comment by BernieHuang2008** [comment]
binding on two div isn't a beautiful code, so i suggested bind on `body` and check whether a image preview is opened.

**Comment by BernieHuang2008** [comment]
hey @bonswouar ! i've found a solution !

what you need to do is add `tabindex=-1` to the `.detail` element and the handler will work!
like this:
![image](https://github.com/searxng/searxng/assets/88757735/5958a5aa-becf-41ca-8c20-77c84524527b)

don't know why, and the solution isn't beautiful, but it actually worked!

**Comment by return42** [comment]
AFAIK this issue should be fixed in

- https://github.com/searxng/searxng/pull/2724#issuecomment-1712518826
